File: 01_Backend/backend/src/main/java/org/example/backend/service/admin/AdminPdReviewService.java
package org.example.backend.service.admin;

import org.example.backend.model.entity.admin.AdminPdReview;
import org.example.backend.repository.admin.AdminPdReviewRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class AdminPdReviewService {

    @Autowired
    AdminPdReviewRepository adminPdReviewRepository;

    //    TODO: 전체조회(read)
    public List<AdminPdReview> findAll() {
        List<AdminPdReview> list = adminPdReviewRepository.findAll();
        return list;
    }

    //    페이징 처리
    public Page<AdminPdReview> findAllByAdminReviewTitleContaining(String reviewTitle,
                                                               Pageable pageable) {
        Page<AdminPdReview> page
                = adminPdReviewRepository
                .findAllByAdminPdReviewTitleContaining(reviewTitle, pageable);
        return page;
    }

    //    상세조회
    public Optional<AdminPdReview> findById(Long reviewId) {
        Optional<AdminPdReview> optionalAdminPdReview
                = adminPdReviewRepository.findById(reviewId);
        return optionalAdminPdReview;
    }

    //    TODO: 등록(insert),수정(update)
    public AdminPdReview save(AdminPdReview adminPdReview) {
        AdminPdReview adminPdReview2 = adminPdReviewRepository.save(adminPdReview);

        return adminPdReview2;
    }

    //    TODO: 삭제(delete)
    public boolean removeById(Long reviewId) {
        if (adminPdReviewRepository.existsById(reviewId) == true) {
            adminPdReviewRepository.deleteById(reviewId);
            return true;
        } else {
            return false;
        }
    }
}

File: frontend-vue/src/services/admin/AdminPdReviewService.js
import http from "@/utils/http-common";
import LoginHeader from "../login/LoginHeader";

class AdminPdReviewService {
    // 전체조회
    getAll(reviewTitle, page, size) {
        return http.get(`/admin/admin-pdqna-review?reviewTitle=${reviewTitle}&page=${page}&size=${size}`,{
            headers: LoginHeader()
        });
    }
    // 상세조회
    get(reviewId) {
        return http.get(`/admin/admin-pdqna-review/${reviewId}`,{
            headers: LoginHeader()
        });
    }
    // 상품리뷰등록함수
    create(data) {
        return http.post("/admin/admin-pdqna-review", data,{
            headers: LoginHeader()
        });
    }
    // 상품리뷰수정함수
    update(reviewId, data) {
        return http.put(`/admin/admin-pdqna-review/${reviewId}`, data,{
            headers: LoginHeader()
        });
    }
    // 상품리뷰삭제함수
    delete(reviewId) {
        return http.delete(`/admin/admin-pdqna-review/deletion/${reviewId}`,{
            headers: LoginHeader()
        });
    }
}

export default new AdminPdReviewService();